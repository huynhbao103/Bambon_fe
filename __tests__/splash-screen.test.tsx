import React from "react";
import { render } from "@testing-library/react-native";
import SplashScreen from "../app/splash-screen";
import { Animated, ImageBackground } from "react-native";

jest.spyOn(Animated, "timing").mockImplementation(() => ({
    start: jest.fn(),
    stop: jest.fn(),
    reset: jest.fn(),
  }));
  
describe("SplashScreen", () => {
    it("starts fade-in animation with correct duration", () => {
      render(<SplashScreen />);
  
      expect(Animated.timing).toHaveBeenCalledWith(expect.any(Object), {
        toValue: 1,
        duration: 2000,
        useNativeDriver: true,
      });
    });
    it("renders splash image correctly", () => {
        const { toJSON } = render(<SplashScreen />);
        expect(toJSON()).toMatchSnapshot();
    });
}); 