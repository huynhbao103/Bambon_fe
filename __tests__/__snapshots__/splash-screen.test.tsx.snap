// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders splash image correctly 1`] = `
<View
  className="flex-1 bg-green-300 justify-center items-center"
  testID="splash-container"
>
  <Image
    collapsable={false}
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/images/splash.png",
      }
    }
    style={
      {
        "height": "100%",
        "opacity": 0,
        "width": "100%",
      }
    }
    testID="splash-image"
  />
</View>
`;
